src/v/cluster/partition_balancer_planner.cc

```diff
@@ -865,6 +867,42 @@ void partition_balancer_planner::get_rack_constraint_repair_actions(
     }
 }
 
+/**
+ * This is the place where we decide about the order in which partitions will be
+ * moved in the case when node disk is being full.
+ */
+size_t partition_balancer_planner::calculate_full_disk_partition_move_priority(
+  model::node_id node_id,
+  const reassignable_partition& p,
+  const request_context& ctx) {
+    static constexpr size_t min_priority = 0;
+    static constexpr size_t max_priority = 1000000;
+
+    if (p.size_bytes() < ctx.config().min_partition_size_threshold) {
+        return min_priority;
+    }
+    /**
+     * Use lowest priority + 1 for redpanda internal topics, this way large
+     * internal topics will be moved before smallest user partitions
+     */
+    if (
+      p.ntp().ns == model::kafka_internal_namespace
+      || p.ntp().tp.topic == model::kafka_consumer_offsets_topic) {
+        return min_priority + 1;
```

Comment:
Same here, I think we should make a distinction for internal topics as well, we should probably use the same formula as for usage partitions, just with different priority "tiers".